eng/common/scripts/Verify-AgentOS.ps1

```diff
@@ -0,0 +1,34 @@
+param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $AgentPool,
+
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $AgentImage
+)
+
+$lookup = @(
+    @{ AgentPool = "Azure Pipelines"; AgentImage = "windows-2019"; AgentOSMatch = $IsWindows }
```

Comment:
Do you think it is better to maintain a full list or should we support some basic string matching? On the image names?

Reply:
I think that this ends up being more compact.

Reply:
What I worry about here is that if we ever need to use a new image or pool we will have to first update this list in eng/common before we can consume that. I'd like to try and avoid that requirement for the normal cases.

Couldn't we do some pattern matching? 
```
AgentImage -match "windows|win|MMS2019" -> IsWindows?
AgentImage -match "ubuntu" -> IsLinux
AgentImage -match "macos" -> IsMac?
```
That would enable future proofing without requiring that we explicitly list all images. Also I'm not sure if the pool name is very interesting for this check so we could eliminate that.

Reply:
Just pushed an update. I'm not sure if we should fail if we see an image that doesn't match a string we expect. I'm inclined not to.